drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -524,312 +500,327 @@
+        following operation:
+          <list style="bullets">
+            <t>AAD = (ICV field || node's immutable IOAM-Data-Fields)</t>
+            <t>ICV = GMAC(AAD)</t>
+          </list>
         </t>
-      </list>
-      </t>
 
-      <t>The method assumes that symmetric keys have been distributed to
-      the respective nodes as well as the Validator(s). The details of the
-      mechanisms used for key distribution are outside the scope of this
-      document.</t>
+        <t>The transit node updates the ICV field in the IOAM Integrity
+        Protection header. Reminder: it MUST always be a 16-octet ICV.</t>
+
+        <t>If the transit node does not add any IOAM-Data-Fields (e.g., it
+        only modifies mutable IOAM-Data-Fields or does nothing), then the
+        transit node MUST NOT update the ICV field in the IOAM Integrity
+        Protection header.</t>
+      </section>
+
+      <section anchor="IPM-Decap" title="Decapsulating node">
+        <t>In the context of integrity protection, the decapsulating node can
+        either act as a transit node (see <xref target="IPM-Transit"/> above) or
+        directly as the Validator (see <xref target="IPM-Validator"/> below).
+        The former is an alternative to put the Validator out of the forwarding
+        path in case of performance concerns.</t>
+      </section>
+
+      <section anchor="IPM-Validator" title="Validator">
+        <t>A node that performs the validation of the integrity protection is
+        referred to as the Validator. This method assumes that symmetric keys
+        have been distributed to the Validator only. The details of the
+        mechanisms used for key distribution are outside the scope of this
+        document. We refer to the method as "Zero Trust" in the sense that
+        neither the encapsulating node, nor transit nodes, nor any other
+        non-IOAM nodes need to be trusted. Indeed, the Validator is the only
+        point of trust, meaning the method is considered a full integrity
+        protection of IOAM-Data-Fields.</t>
+
+        <t>The Validator MUST recompute the ICV by iteratively following the
+        previous steps of the method in the same order, using the respective
+        symmetric keys received previously. The recomputed ICV is then compared
+        to the received ICV field. As a result, the Validator can detect if the
+        integrity of IOAM-Data-Fields is intact or altered. The validation is
+        one-step in some cases (e.g., with POT Type-0 or E2E), where only the
+        encapsulating node updates the ICV, according to the definition of this
+        method. For other cases where transit nodes also update the ICV (e.g.,
+        with Trace Option-Types), the Validator MUST identify these transit
+        nodes in order to look up their respective keys. For that, something
+        unique representing a node MUST be included in IOAM-Data-Fields (e.g.,
```

Comment:
s/For that, something unique representing a node MUST be included in IOAM-Data-Fields (e.g., the "node_id" for Trace Option-Types)./For that, a unique identifier of the node, such as the "node_id" for Trace Option-Types MUST be included in IOAM-Data-Fields/